gravity/filters/delete_dml_column_filter.go

```diff
@@ -49,6 +55,23 @@ func (f *deleteDmlColumnFilter) Filter(msg *core.Msg) (continueNext bool, err er
 		return false, errors.Errorf("DmlMsg is null")
 	}
 
+	if msg.DmlMsg.PkColumns != nil {
+		var newPkColumns []string
+		for _, oldName := range msg.DmlMsg.PkColumns {
+			for _, name := range f.columns {
+				_, ok := msg.DmlMsg.Pks[name]
```

Comment:
why check `Pks` again? Can't we remove it directly? `Pks` and `PkColumns` should always in sync I think.

Reply:
It's not easy to delete an element in a slice by the element's content.

Reply:
I mean `_, ok := msg.DmlMsg.Pks[name]` seems unnecessary.

Reply:
if `f.columns` have column name that is not a primary key, then there will be a problem

Reply:
don't quite understand. if `f.columns` doesn't contain `oldName`, then it should be kept, is it right?

Reply:
what if `columns` is ['pk2', 'pk1', 'r'], `PkColumns` is ['pk1', 'pk2', 'pk3'], then `newPkColumns` would be ['pk1', 'pk2']